### ISSUE TITLE
Conditional Field Validation

### ISSUE THREAD

**Issue by garthvh** [meshtastic/design#9]
Would it be good to have common field validation across platforms so that conflicting settings are not enabled by users? 

* Encryption and JSON mode in MQTT Can not be enabled at the same time
* Client proxy and JSON mode can not be enabled at the same time
* Both phone and Device GPS should not be enabled at the same time.
* Wifi, MQTT and Client Proxy should not be enabled at the same time
* Sections with client and sender enabled options (detection sensor, range test) should not expose settings not related to that device's role.

**Comment by b8b8** [comment]
"Both phone and Device GPS should not be enabled at the same time."

Is there not a legitimate use case for this though. If you have higher quality phone GPS use it, but then default back to node GPS automatically. My car node (when i am actively driving around) is bluetoothed to my phone and providing high quality GPS updates to the device. When i am parked or someone else is using my car that doesnt have meshtastic, the device will use the onboard GPS and act as a "tracker" around town.

**Comment by garthvh** [comment]
> Is there not a legitimate use case for this though. If you have higher quality phone GPS use it, but then default back to node GPS automatically. My car node (when i am actively driving around) is bluetoothed to my phone and providing high quality GPS updates to the device. When i am parked or someone else is using my car that doesnt have meshtastic, the device will use the onboard GPS and act as a "tracker" around town.

Yeah this one is more complicated for sure.  Smart position actually works really well, you can prevent blended positions pretty easily and fall back to device gps when the phone disconnects, but all the other modes just blend the data together.

**Comment by b8b8** [comment]
Blending is good. Just dont want a situation where it sets to phone position, toggles off discrete GPS, then loses phone position and doesnt turn discrete GPS on again.

**Comment by garthvh** [comment]
Status update (August 2026): this list has been partially overtaken by events — JSON mode no longer exists (the deprecated toggle was removed from the clients), iOS hides advanced LoRa settings when a preset is in use, and several other rules are enforced screen-by-screen rather than by a shared validation layer. If cross-client validation is still wanted, the next step is refreshing the rule list against 2.7/2.8 reality and deciding which rules belong in firmware (authoritative) vs client UX (advisory). Otherwise this can close as absorbed into the per-config screens.
